src/Type/Php/RandomIntFunctionReturnTypeExtension.php

```diff
@@ -16,11 +17,15 @@ class RandomIntFunctionReturnTypeExtension implements \PHPStan\Type\DynamicFunct
 
 	public function isFunctionSupported(FunctionReflection $functionReflection): bool
 	{
-		return $functionReflection->getName() === 'random_int';
+		return in_array($functionReflection->getName(), ['random_int', 'rand']);
 	}
 
 	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
 	{
+		if ($functionReflection->getName() === 'rand' && count($functionCall->args) === 0) {
```

Comment:
because of `ParametersAcceptorSelector::selectSingle` which throw'ed an Exception, I removed the `'rand\'1' => ['int']` from the functionMap and moved the corresponding logic directly into the extension.

that way I was able to re-use the extension without further changes

Reply:
You should actually call `selectFromArgs`.

Reply:
since adding this IF the `selectSingle` later on is enough. should I nevertheless add `selectFromArgs`?

Reply:
I don't get it. You removed `selectSingle` because it was throwing an exception, but there's still `selectSingle` in this extension. I think it would be thrown if someone called `rand(1)`.

Basically you should always use `selectFromArgs` if you have `args` available. It's the safest way.

Reply:
`rand(1)` is not valid. see https://3v4l.org/HYrTe

nevertheless, I changed it to `selectFromArgs`